packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -588,9 +616,42 @@ export class ApexDebug extends DebugSession {
     }
     if (message.sobject.Description) {
       logMessage += ` | ${message.sobject.Description}`;
+      const regExp: RegExp = /^(.*)\[(\d+)\]\|/;
+      const matches = message.sobject.Description.match(regExp);
+      if (matches && matches.length === 3) {
+        const possibleClassName = matches[1];
+        const possibleClassLine = parseInt(matches[2]);
+        const possibleSourcePath = this.myBreakpointService.getSourcePathFromPartialTyperef(
+          possibleClassName
+        );
+        if (possibleSourcePath) {
+          eventDescriptionSourceFile = new Source(
```

Comment:
Apex Debugger event type 'Debug' has a description in the form of '<classname>[line number]|<actual message>'. The debug protocol has recently supported including a source location in an output event, so I'm parsing that stuff here. The idea is to link a debug console message so user can quickly navigate to the Apex source. The VS Code UI doesn't actually support it yet, but including it now is harmless because it's ignored. The frontend support seems to be planned for this month: `https://github.com/Microsoft/vscode/issues/24490`